docs: update docs website (#362)
Azure/secrets-store-csi-driver-provider-azure: website/content/en/configurations/identity-access-modes/system-assigned-msi-mode.md

---
type: docs
title: "System-assigned Managed Identity"
linkTitle: "System-assigned Managed Identity"
weight: 4
description: >
  Use a System-assigned Managed Identity to access Keyvault.
---

> Supported with Linux and Windows

<details>
<summary>Examples</summary>

- `SecretProviderClass`
```yaml

> # This is SecretProviderClass example using system-assigned identity for authentication with Key Vault

Reviewer: ```suggestion
# This is a SecretProviderClass example using system-assigned identity for authentication with Key Vault

```

apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
kind: SecretProviderClass
metadata:
  name: azure-kvname-system-msi
spec:
  provider: azure
  parameters:
    usePodIdentity: "false"
    useVMManagedIdentity: "true"
    userAssignedIdentityID: ""      # If empty, then defaults to use the system assigned identity on the VM
    keyvaultName: "kvname"
    cloudName: ""                   # [OPTIONAL for Azure] if not provided, azure environment will default to AzurePublicCloud
    objects:  |
      array:
        - |
          objectName: secret1
          objectType: secret        # object types: secret, key or cert
          objectVersion: ""         # [OPTIONAL] object versions, default to latest if empty
        - |
          objectName: key1
          objectType: key
          objectVersion: ""
    tenantId: "tid"                 # the tenant ID of the KeyVault  
``` 

- `Pod` yaml
```yaml

> # This is a sample pod definition for using SecretProviderClass and system-assigned identity for authentication with Key Vault

Reviewer: ```suggestion
# This is a sample pod definition for using SecretProviderClass and system-assigned identity to access Key Vault

```

kind: Pod
apiVersion: v1
metadata:
  name: nginx-secrets-store-inline-system-msi
spec:
  containers:
    - name: nginx
      image: nginx
      volumeMounts:
      - name: secrets-store01-inline
        mountPath: "/mnt/secrets-store"
        readOnly: true
  volumes:
    - name: secrets-store01-inline
      csi:
        driver: secrets-store.csi.k8s.io
        readOnly: true
        volumeAttributes:
          secretProviderClass: "azure-kvname-system-msi"
```
</details>

## Configure System-assigned Managed Identity to access Keyvault

> AKS uses system-assigned managed identity as [cluster managed identity](https://docs.microsoft.com/en-us/azure/aks/use-managed-identity). This managed identity shouldn't be used to authenticate with KeyVault. You should consider using a [User-assigned managed identity](./user-assigned-msi-mode) instead.

Reviewer: ```suggestion
AKS uses system-assigned managed identity as [cluster managed identity](https://docs.microsoft.com/en-us/azure/aks/use-managed-identity). This managed identity shouldn't be used to access KeyVault. You should consider using a [User-assigned managed identity](./user-assigned-msi-mode) instead.

```
